src/main/java/org/apache/sysml/runtime/instructions/cp/ListObject.java

```diff
@@ -81,7 +97,34 @@ public Data slice(String name) {
 		//return existing entry
 		return slice(pos);
 	}
-	
+
+	public Data slice(String name, ValueType vt, DataType dt) {
```

Comment:
This method should not be necessary because the function class does any necessary value type casts automatically. So, let's please neither duplicate this logic nor create unnecessary copies here.